.github/workflows/release.yaml

```
@@ -0,0 +1,62 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(yarn -s run get-version)" = "${{ inputs.tag }}"
+      - name: Give user instruction
+        if: failure()
+        run: echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag ${{ inputs.tag }} must match the version $(yarn -s run get-version)"
+      # Setup .npmrc file to publish to GitHub Packages
+      - name: Configure node with GitHub Packages
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
+      - run: yarn install --frozen-lockfile
+      - name: Publish to GitHub Packages
+        run: npm publish --provenance
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Configure node with npmjs.org
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Publish to npmjs.org
+        run: npm publish --provenance
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

would be better to have a single `node` setup stage, then publish to each registry:

```suggestion
      - name: Publish to GitHub Packages
        run: npm publish --provenance
        env:
          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      - name: Publish to npmjs.org
        run: |
          npm config set registry https://registry.npmjs.org/
          npm publish --provenance
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

---

fixed it by specifying `--registry` when calling `npm publish`

---

Turns out I **need** to setup two nodes, just because they each go and configure a `.npmrc` in a proper way for me to authenticate with a npm registry. In other words this is what they do:

For example if I set `registry-url: https://npm.pkg.github.com`:
```
# .npmrc
//npm.pkg.github.com/:_authToken=${NODE_AUTH_TOKEN}
@scality:registry=https://npm.pkg.github.com/
always-auth=false
```